dev/tools/gen_keycodes/lib/utils.dart

```diff
@@ -108,3 +116,140 @@ String toHex(int value, {int digits = 8}) {
 int getHex(String input) {
   return int.parse(input, radix: 16);
 }
+
+/// Given an [input] string, wraps the text at 80 characters and prepends each
+/// line with the [prefix] string. Use for generated comments.
+String wrapString(String input, {required String prefix}) {
+  final int wrapWidth = 80 - prefix.length;
+  final StringBuffer result = StringBuffer();
+  final List<String> words = input.split(RegExp(r'\s+'));
+  String currentLine = words.removeAt(0);
+  for (final String word in words) {
+    if ((currentLine.length + word.length) < wrapWidth) {
+      currentLine += ' $word';
+    } else {
+      result.writeln('$prefix$currentLine');
+      currentLine = word;
+    }
+  }
+  if (currentLine.isNotEmpty) {
+    result.writeln('$prefix$currentLine');
+  }
+  return result.toString();
+}
+
+/// Run `fn` with each corresponding element from list1 and list2.
+///
+/// If `list1` has a different length from `list2`, the execution is aborted
+/// after printing an error.
+///
+/// An null list is considered a list with length 0.
+void zipStrict<T1, T2>(Iterable<T1> list1, Iterable<T2> list2, void Function(T1, T2) fn) {
+  if (list1 == null && list2 == null)
+    return;
+  if (list1.length != list2.length) {
+    print('Mismatched lists $list1 to $list2');
```

Comment:
Maybe throw or assert instead of printing?

Reply:
Thanks for pointing out. Yes. This was an overlooked case when I convert the script to asserts.